relay-server/src/endpoints/spans.rs

```diff
@@ -1,37 +1,58 @@
 use axum::extract::{DefaultBodyLimit, Json};
-use axum::http::StatusCode;
+use axum::http::{Request, StatusCode};
 use axum::response::IntoResponse;
 use axum::routing::{post, MethodRouter};
+use axum::RequestExt;
+use axum_extra::protobuf::Protobuf;
 use bytes::Bytes;
 
 use relay_config::Config;
-use relay_spans::TracesData;
+use relay_spans::OtelTrace::TracesData;
 
-use crate::endpoints::common::{self, BadStoreRequest};
+use crate::endpoints::common;
 use crate::envelope::{ContentType, Envelope, Item, ItemType};
-use crate::extractors::RequestMeta;
+use crate::extractors::{RawContentType, RequestMeta};
 use crate::service::ServiceState;
 
-async fn handle(
+async fn handle<B>(
     state: ServiceState,
+    content_type: RawContentType,
     meta: RequestMeta,
-    Json(trace): Json<TracesData>,
-) -> Result<impl IntoResponse, BadStoreRequest> {
-    let mut envelope = Envelope::from_request(None, meta);
-    for resource_span in trace.resource_spans {
-        for scope_span in resource_span.scope_spans {
-            for span in scope_span.spans {
-                let Ok(payload) = serde_json::to_vec(&span) else {
-                    continue;
-                };
-                let mut item = Item::new(ItemType::OtelSpan);
-                item.set_payload(ContentType::Json, payload);
-                envelope.add_item(item);
+    request: Request<B>,
+) -> axum::response::Result<impl IntoResponse>
+where
+    B: axum::body::HttpBody + Send + 'static,
+    B::Data: Send + Into<Bytes>,
+    B::Error: Into<axum::BoxError>,
+{
+    let mut trace = None;
+    if content_type.as_ref().starts_with("application/json") {
+        let json: Json<TracesData> = request.extract().await?;
+        trace = Some(json.0)
+    } else if content_type.as_ref().starts_with("application/x-protobuf") {
+        let protobuf: Protobuf<TracesData> = request.extract().await?;
+        trace = Some(protobuf.0)
+    }
```

Comment:
I think this should work:
```suggestion
    let trace = if content_type.as_ref().starts_with("application/json") {
        let json: Json<TracesData> = request.extract().await?;
        Some(json.0)
    } else if content_type.as_ref().starts_with("application/x-protobuf") {
        let protobuf: Protobuf<TracesData> = request.extract().await?;
        Some(protobuf.0)
    };
```